packages/components/grid/src/filter.scss

```diff
@@ -27,24 +28,24 @@ sl-popover {
 
 header {
   align-items: center;
+  background: var(--sl-elevation-surface-raised-default);
   border-block-end: 1px solid var(--sl-color-elevation-border-raised);
   display: flex;
+  font-weight: var(--sl-font-weight-semibold);
   gap: 0.5rem;
+  inset-block-start: calc(var(--sl-size-300) * -1);
   justify-content: space-between;
-  padding-block: 0.25rem;
-  padding-inline: 0.5rem;
+  margin-block: calc(var(--sl-size-300) * -1) var(--sl-size-200);
+  padding-block-start: var(--sl-size-300);
+  position: sticky;
+  z-index: 1;
```

Comment:
I've started documenting the value of `z-index`, so it's clear why it is a certain value.